src/poetry/installation/wheel_installer.py

```diff
@@ -39,16 +39,35 @@ def write_to_fs(
         stream: BinaryIO,
         is_executable: bool,
     ) -> RecordEntry:
+        import base64
+        import hashlib
+
         from installer.records import Hash
         from installer.records import RecordEntry
+        from installer.utils import _COPY_BUFSIZE
         from installer.utils import copyfileobj_with_hashing
         from installer.utils import make_file_executable
 
         target_path = Path(self.scheme_dict[scheme]) / path
         if target_path.exists():
             # Contrary to the base library we don't raise an error here since it can
             # break pkgutil-style and pkg_resource-style namespace packages.
-            logger.warning(f"Installing {target_path} over existing file")
+            # We instead log a warning and ignore it. See issue #9158.
+            logger.warning(f"{target_path} already exists. Ignoring.")
+            # Following is copied from copyfileobj_with_hashing, to calculate the hash
```

Comment:
this is basically the same as `_get_file_hash()` in builders/editable.py, wibni this was factored out into a utility

Reply:
Done!